### PR TITLE
Added step for draft release and sanity testing

### PR DESCRIPTION
**Pull request by shrutiburman**
# Fixes [DII-181](https://issues.corp.twilio.com/browse/DII-181)

This PR:
Includes the step to perform basic sanity testing with npm installations.
Does a draft release with tag "version-draft" before the actual release. 

Github actions used: 
[delete-tag-and-release](https://github.com/dev-drprasad/delete-tag-and-release)
[undo-push](https://github.com/marketplace/actions/undo-push)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,6 +12,9 @@ on:
       homebrew-prerelease:
         description: 'HomeBrew prerelease'
         default: 'false'
+      draft-release:
```
Comment (shrutiburman) [inline .github/workflows/release.yml:15]: Input parameter flag to use the same workflow for a draft release and then an actual release.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -55,12 +58,27 @@ jobs:
         run: |
           git pull
           npm install
-      - name: Semantic Release runs
+      - name: Semantic Release runs for draft release
+        if: ${{github.event.inputs.draft-release == 'true'}}
         id: semantic-release
+        run: npx semantic-release -t \${version}-draft
```
Comment (shrutiburman) [inline .github/workflows/release.yml:64]: If the workflow is triggered for draft-release, tag would be `version-draft`

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -69,7 +87,32 @@ jobs:
           TAG_NAME: ${{steps.semantic-release.outputs.TAG_NAME}}
           RELEASE_BODY: ${{github.event.inputs.change-log}}
           REPO_NAME: twilio/twilio-cli
+  delete-draft-release:
```
Comment (shrutiburman) [inline .github/workflows/release.yml:90]: This is cleanup step to:
1. Delete the tagged (draft) release.
2. Revert the last commit made by semantic-bot, which has changelog.

File: .github/workflows/cli-test-npm.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: CLI Tests on npm installations
+on:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
```
Comment (ravali-rimmalapudi) [inline .github/workflows/cli-test-npm.yml:7]: Don't we need to add testing for windows and mac?

File: .github/workflows/cli-test-npm.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/setup-node@v2
+        with:
+         node-version: ${{ matrix.node-version }}
+      -  name: Global - Testing installation of Twilio CLI (1)
+         run: |
+          npm install -g twilio-cli
+          node .github/asserts/validate_tests.js
+      - name: Global - Testing upgrade Twilio CLI (2)
+        run: |
+         npm install -g twilio-cli@{previous-version}
```
Comment (ravali-rimmalapudi) [inline .github/workflows/cli-test-npm.yml:22]: From where we are getting `previous-version` value?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -55,12 +58,27 @@ jobs:
...
           git pull
           npm install
-      - name: Semantic Release runs
+      - name: Semantic Release runs for draft release
+        if: ${{github.event.inputs.draft-release == 'true'}}
         id: semantic-release
+        run: npx semantic-release -t \${version}-draft
+        env:
+          GITHUB_TOKEN: ${{ secrets.REPO_ACCESS_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (ravali-rimmalapudi) [inline .github/workflows/release.yml:67]: Are we planning to push the draft release tag also to main npmjs? and draft tag will be released as normal release.

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -69,7 +87,32 @@ jobs:
...
+  delete-draft-release:
+    if: ${{github.event.inputs.draft-release == 'true'}}
+    runs-on: ubuntu-latest
+    needs: [draft-release]
+    steps:
+      - name: Delete draft release
+      - uses: dev-drprasad/delete-tag-and-release@v0.2.0
+        with:
+          delete_release: true
+          tag_name: \${version}-draft
```
Comment (ravali-rimmalapudi) [inline .github/workflows/release.yml:99]: From where we are getting the `version` value?

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -69,7 +87,32 @@ jobs:
...
+      - name: Extract branch name
+        id: extract_branch
+        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+           ref: ${{ steps.extract_branch.outputs.branch }}
+           fetch-depth: 0 
+      - name: Undo Push
+        uses: exions/undo-push@v1
```
Comment (ravali-rimmalapudi) [inline .github/workflows/release.yml:111]: Is it revert the latest commit on the branch?

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -111,6 +155,7 @@ jobs:
           REPO_ACCESS_TOKEN: ${{ secrets.REPO_ACCESS_TOKEN }}
           BRANCH_NAME: ${{steps.extract_branch.outputs.branch}}
   platform-executables-release:
+    if: ${{github.event.inputs.draft-release != 'true'}}
```
Comment (ravali-rimmalapudi) [inline .github/workflows/release.yml:158]: Can you please add the local workflow run link?

**Pull request closed without merging**